### ISSUE TITLE
Upload created tmp file and uploadurl not showing

### ISSUE THREAD

**Issue by karinnahmay** [krizalys/onedrive-php-sdk#69]
@krizalys hi, this actually worked but i got ~tmpxxx file created inside my folder. How can i get rid of that?
i used this exact code:

```
$stream = fopen($path_to_file_being_uploaded, 'rb');
$upload = $folder->startUpload(basename($path_to_file_being_uploaded), $stream);
```

And after complete upload i would like to get the upload URL but no luck:

```
$driveItem = $upload->complete();
$permission = $upload->uploadUrl;
```

i thought i would get the URL by `$driveItem->uploadUrl`;

but still no luck. 

Thanks!

**Comment by krizalys** [comment]
> i got ~tmpxxx file created inside my folder. How can i get rid of that?

@karinnahmay This could be the result of your PHP configuration. In some cases, memory streams are backed by temporary files. In some other cases, this kind of files are created when reading files uploaded from a browser via a web form.

For more details on these, please look into the relevant pages of PHP documentation:

- https://www.php.net/manual/en/wrappers.php.php
- https://www.php.net/manual/en/features.file-upload.php

> And after complete upload i would like to get the upload URL but no luck:

`DriveItemProxy::uploadURL` is the URL where you are can (re)upload the file. You probably can ignore it because the library knows what URLs to use internally.

There are 2 kinds of URLs for a OneDrive file:
1. A "graph" URL, which the URL of the file as known on Microsoft Graph - needs login and access to the file - this URL would be `https://graph.microsoft.com/v1.0/drives/{drive-id}/items/{item-id}` or `https://graph.microsoft.com/v1.0/drives/{drive-id}/root:/{item-path}` (where `{item-path}` would be built from `$folder` path and `basename($path_to_file_being_uploaded)`)
2. A "shared" URL that people can access without having access to your account or browsing through the graph - you need to first explicitly create a link to the item using `DriveItemProxy::createLink()`: the returned value is a `PermissionProxy` so you can get the URL using an expression like `$permission->link->webUrl`

**Comment by karinnahmay** [comment]
Hi @krizalys,

Unfortunately Shared link is not generated, here is my code:

```
  $stream = fopen($path_file, 'rb');
  $upload = $dataItem->startUpload(basename($path_file), $stream);
  $driveItem = $upload->complete();
  $permission = $dataItem->createLink("view");
```

can you please help or provide instructions on why this happened? Thanks sir!

**Comment by krizalys** [comment]
@karinnahmay Please see a possible fix: #70.
